test/integration/contractor_pays_project_test.rb

```diff
@@ -0,0 +1,46 @@
+require 'test_helper'
+
+class ContractorPaysProjectTest < ActionDispatch::IntegrationTest
+  test "contractor pays project" do
+    devs = create_list(:developer, 2)
+    contractor = create(:contractor)
+    ApplicationController.any_instance.stubs(:current_contractor).returns(contractor)
+    project = create(:project, contractor: contractor, developers:devs)
+
+    assert_equal project.status, "Pending"
+    devs.each do |dev|
+      assert_equal dev.status, "available"
+    end
+
+    visit contractor_project_path(project)
+    click_on "Submit Payment"
+    click_on "Make Payment"
+
+    project.reload
+    assert_equal contractor_project_path(project), current_path
+    assert_equal "Paid & Active", project.status
+    devs.each do |dev|
+      dev.reload
+      assert_equal "unavailable", dev.status
+    end
+  end
+
+  test "contractor cannot pay project with unavailable developers" do
+    devs = create_list(:developer, 2)
+    busy_dev = create(:developer, status: 'unavailable')
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.